src/vim9execute.c

```diff
@@ -4077,9 +4094,17 @@ exec_instructions(ectx_T *ectx)
 			case EXPR_GEQUAL: res = arg1 >= arg2; break;
 			case EXPR_SMALLER: res = arg1 < arg2; break;
 			case EXPR_SEQUAL: res = arg1 <= arg2; break;
+			case EXPR_LSHIFT: res = (uvarnumber_T)arg1 << arg2;
+					  break;
+			case EXPR_RSHIFT: res = (uvarnumber_T)arg1 >> arg2;
+					  break;
 			default: break;
 		    }
 
+		    if (iptr->isn_arg.op.op_type == EXPR_RSHIFT)
+			// clear the topmost sign bit
+			res &= ~((uvarnumber_T)1 << MAX_LSHIFT_BITS);
```

Comment:
Have you tried it without this post shift fixup? Since the type of the the left hand side is unsigned, no adjustment is needed.